### PR TITLE
Add mask to ui package

### PR DESCRIPTION
**Pull request by nturinski**

### CODE REVIEW

File: ui/test/callWithTelemetryAndErrorHandling.test.ts [hunk 1]
```diff
@@ -63,4 +63,14 @@ suite('callWithTelemetryAndErrorHandling tests', () => {
             }),
             /testFuncErrorAsync/);
     });
+
+    test('mask', async () => {
```
Comment (EricJizbaMSFT) [inline ui/test/callWithTelemetryAndErrorHandling.test.ts]: You should copy over all unit tests: https://github.com/microsoft/vscode-azuretools/blob/master/appservice/test/callWithMaskHandling.test.ts

File: ui/src/callWithTelemetryAndErrorHandling.ts [hunk 2]
```diff
@@ -67,6 +67,20 @@ export async function callWithTelemetryAndErrorHandling<T>(callbackId: string, c
     }
 }
 
+export async function callWithMaskHandling<T>(callback: () => Promise<T>, valueToMask: string): Promise<T> {
```
Comment (EricJizbaMSFT) [inline ui/src/callWithTelemetryAndErrorHandling.ts]: Mmm I'd prefer a separate file for this

File: ui/src/callWithTelemetryAndErrorHandling.ts [hunk 3]
```diff
@@ -67,6 +67,20 @@ export async function callWithTelemetryAndErrorHandling<T>(callbackId: string, c
...
+    try {
+        return await callback();
+    } catch (error) {
+        const parsedError: IParsedError = parseError(error);
+
+        if (parsedError.isUserCancelledError) {
+            throw error;
+        }
+
+        throw new Error(parsedError.message.replace(new RegExp(valueToMask, 'g'), '***'));
```
Comment (EricJizbaMSFT) [inline ui/src/callWithTelemetryAndErrorHandling.ts]: I don't think a RegExp will work if `valueToMask` has special characters

File: ui/test/callWithTelemetryAndErrorHandling.test.ts (hunk 1 above)
Comment (nturinski) [reply ui/test/callWithTelemetryAndErrorHandling.test.ts]: Oh man, I didn't expect me to actually have unit tests for this. Hooray past Nathan (and past Eric who probably made him make unit tests)

File: ui/test/callWithMaskHandling.test.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                    throw new Error(errorMessage);
+                }, credentials);
+            }, (err) => {
+                return validateError(err, credentials);
+            }, 'Credentials were not properly masked from error string');
+        });
+
+        test("Value masked (single occurance) with thrown error", async () => {
+
+            // tslint:disable-next-line: no-multiline-string
```
Comment (EricJizbaMSFT) [inline ui/test/callWithMaskHandling.test.ts]: I can't tell the difference between this test and the first test

Comment (EricJizbaMSFT) [reply ui/test/callWithMaskHandling.test.ts]: Can you add a test for the special characters case I mentioned elsewhere

Comment (nturinski) [reply ui/test/callWithMaskHandling.test.ts]: I'm not sure why I did it, but once has `throw new Error(errorMessage)` versus just `throw errorMessage`

**Pull request merged**